src/main/java/com/amplitude/api/DeviceInfo.java

```diff
@@ -194,6 +196,9 @@ private String getLanguage() {
         }
 
         private String getAdvertisingId() {
+            if (DeviceInfo.customAdvertisingId != null) {
+                return DeviceInfo.customAdvertisingId;
```

Comment:
Thanks for the contribution! Maybe this could be on client level other than in DeviceInfo? Multiple instances may cause conflict I guess.